src/components/ManageAccountsDrawer/components/ImportAccounts.tsx

```diff
@@ -122,41 +127,13 @@ export const ImportAccounts = ({ chainId, onClose }: ImportAccountsProps) => {
   // But also need to remove accounts that were disabled.
 
   // initial fetch to detect the number of accounts based on the "first empty account" heuristic
-  useEffect(() => {
-    ;(async () => {
-      if (!wallet) return
-
-      let accountNumber = 0
-      const accounts = []
-
-      while (true) {
-        try {
-          const accountResult = await queryClient.fetchQuery(
-            reactQueries.accountManagement.accountIdWithActivity(accountNumber, chainId, wallet),
-          )
-
-          if (!accountResult) break
-
-          const { accountId, hasActivity } = accountResult
-
-          // Add the account before checking if it has activity so user has ability to toggle it if needed.
-          accounts.push({ accountNumber, accountId })
-
-          // If the account has no activity, it's the first empty account.
-          if (!hasActivity) {
-            break
-          }
-        } catch (error) {
-          console.error(error)
-          break
-        }
-
-        accountNumber++
-      }
```

Comment:
This caused a very nasty race condition where previous queries could arrive after the current one, resulting in the wrong accounts being rendered. Solution was to move into react query so the chainId in the query key could protect us